collectors/framework/framework.go

```diff
@@ -220,6 +222,32 @@ func (a *AvroDatum) transform(nodeInfo collectors.NodeInfo) (producers.MetricsMe
 		return pmm, err
 	}
 
+	decodedDps, ok := dps.([]interface{})
```

Comment:
@darkonie doesn't look that way. I tried it out and i think that type assertion failed (the tests failed at least).  guessing the slice is literally a slice of interface{} to begin with.